format.go

```diff
@@ -98,33 +98,33 @@ type protobufFormat struct{}
 
 // NewDecoder implements Format.
 func (protobufFormat) NewDecoder(r io.Reader) (FormatDecoder, error) {
-	return protobufWrapper{Reader: pbio.NewDelimitedReader(r, 1<<28)}, nil
+	return protobufWrapper{dec: pbio.NewDecoder(r)}, nil
 }
 
 // NewEncoder implements Format.
 func (protobufFormat) NewEncoder(w io.Writer) (FormatEncoder, error) {
-	return protobufWrapper{Writer: pbio.NewDelimitedWriter(w)}, nil
+	return protobufWrapper{enc: pbio.NewEncoder(w)}, nil
 }
 
 type protobufWrapper struct {
-	pbio.Reader
-	pbio.Writer
+	dec *pbio.Decoder
+	enc *pbio.Encoder
 }
 
 func (w protobufWrapper) Decode(v interface{}) error {
 	msg, ok := v.(proto.Message)
 	if !ok {
 		return fmt.Errorf("value %v (%T) is not a proto.Message", v, v)
 	}
-	return w.ReadMsg(msg)
+	return w.dec.Decode(msg)
 }
 
 func (w protobufWrapper) Encode(v interface{}) error {
```

Comment:
actually, I think we can support both, gogo + protobuf v2  right here. you just need to make a `switch msg := v.(type)`